test/parsing.test.ts

```diff
@@ -984,6 +984,34 @@ describe('Zod Parsing', () => {
     });
   });
 
+  it('should collect deeply nested errors as a tree of errors', () => {
+    const deepSchema = z.object({ x: z.string(), y: z.number() }).collectErrors();
+    const schema = z.object({ a: deepSchema, b: deepSchema, c: deepSchema, d: z.string() }).collectErrors();
+    const err = schema.try({ a: { x: 3, y: 4 }, b: { x: '3', y: '4' }, c: { x: '3', y: 4 }, d: 'hello' });
+    assert.ok(err instanceof z.ValidationError);
+    assert.strictEqual(
+      err.message,
+      [
```

Comment:
it seems to me, that it would be ideal if we could join the path so instead of:
```
error parsing object at path: "a" - error parsing object at path: "x" - expected type to be string but got number
```

We would get:
```
error parsing object at path: "a.x" - expected type to be string but got number
```

What do you think?

Reply:
okay, so, trying to do this - it's tricky, since we don't know at the time of ValidationError construction whether or not we are the root node of the tree.
we'll have to store the error message itself separately from the `error parsing object at path: "a.x" ` jazz. Workin on that now

Reply:
okay - spent some time on this. it doesn't look like it's gonna be easy or straightforward, and will require some new object fields, and require me to understand the `myzod` codebase much more deeply to be able to do it right.
What do you think about merging this as-is, and then opening an issue to improve the error messages? :)


What do you think about

Reply:
I'll take a look tomorrow morning, and see if I can't come up with something. I used to know the codebase pretty well! (that's a joke but also true.)

If I can't come up with something by end of week, I will merge as is.

Reply:
sounds good!
I'm thinking that the message for an error will have to recurse into all its children until it finds a leaf node; feels a little expensive.

Reply:
Is there a way we can make it lazy? Like recurse when we want the full deep errors? Or does it need to be a resolved object?

Reply:
was about to go to sleep, and couldn't get that solution off my mind. pushed to this branch!
it's not the most optimal - you could do some fancy caching stuff to avoid duplicating work when calculating the errors of sub-trees, but like, it's in the error case anyways, and only when you turn `collectErrors` on, which is already way slower than not doing it.
should re-run the benchmarks, but IMO this should solve it just fine.
also augmented the test cases to crystallize this behaviour!

Reply:
> Is there a way we can make it lazy? Like recurse when we want the full deep errors? Or does it need to be a resolved object?

this what i was kinda going for with my original implementation, but got way too in-the-weeds. see my above comment (nice timing on yours, basically exactly when i pushed)